karyon3-core/src/main/java/com/netflix/karyon/Karyon.java

```diff
@@ -90,31 +88,34 @@ protected void configureServlets() {
     
     @Singleton
     class KaryonFeatureSetImpl implements KaryonFeatureSet {
-        private final IdentityHashMap<KaryonFeature<?>, Object> features;
+        private final IdentityHashMap<KaryonFeature<?>, Object> featureOverrides;
         
         @Inject
-        private PropertySource properties;
+        private PropertySource properties = DefaultPropertySource.INSTANCE;
         
         @Inject
-        public KaryonFeatureSetImpl(IdentityHashMap<KaryonFeature<?>, Object> features) {
-            this.features = features;
+        public KaryonFeatureSetImpl(IdentityHashMap<KaryonFeature<?>, Object> featureOverrides) {
+            this.featureOverrides = featureOverrides;
         }
         
         @SuppressWarnings("unchecked")
         @Override
         public <T> T get(KaryonFeature<T> feature) {
-            if (features.containsKey(feature)) {
-                return (T) features.get(feature);
-            }
-            else if (properties == null) {
-                return feature.getDefaultValue();
-            }
-            else {
-                return (T) properties.get(feature.getKey(), feature.getType(), feature.getDefaultValue());
-            }
+            return featureOverrides.containsKey(feature)
+                ? (T) featureOverrides.get(feature)
+                : (T) properties.get(feature.getKey(), feature.getType(), feature.getDefaultValue());
         }
     }
 
+    final static Comparator<AutoBinder> byPriority = new Comparator<AutoBinder>() {
+        @Override
+        public int compare(AutoBinder o1, AutoBinder o2) {
+            int p1 = o1.getClass().isAnnotationPresent(Priority.class) ? o1.getClass().getAnnotation(Priority.class).value() : 0;
+            int p2 = o2.getClass().isAnnotationPresent(Priority.class) ? o2.getClass().getAnnotation(Priority.class).value() : 0;
+            return p2 - p1;
```

Comment:
My above comment doesn't seem to apply to the use case here. Seems you only use it to sort AutoBinder so far.